JSQMessagesViewController/Factories/JSQMessagesVideoThumbnailFactory.m

```diff
@@ -0,0 +1,90 @@
+//
+//  JSQMessagesVideoThumbnailFactory.m
+//  JSQMessages
+//
+//  Created by Xi Huang on 8/22/16.
+//  Copyright © 2016 Hexed Bits. All rights reserved.
+//
+
+#import <AVFoundation/AVFoundation.h>
+
+#import "JSQVideoMediaItem.h"
+
+#import "JSQMessagesVideoThumbnailFactory.h"
+#import "JSQMessagesBubbleImageFactory.h"
+
+#import "UIImage+JSQMessages.h"
+
+@implementation JSQMessagesVideoThumbnailFactory
+
++ (UIImageView *)defaultPlayIcon {
+    // same sizing logic with the mediaViewDisplaySize method from JSQMediaItem
+    CGSize size = CGSizeZero;
+    if ([UIDevice currentDevice].userInterfaceIdiom == UIUserInterfaceIdiomPad) {
+        size = CGSizeMake(315.0f, 225.0f);
+    }
+    else {
+        size = CGSizeMake(210.0f, 150.0f);
+    }
+    
+    UIImage *playIcon = [[UIImage jsq_defaultPlayImage] jsq_imageMaskedWithColor:[UIColor lightGrayColor]];
+    
+    UIImageView *playIconImageView = [[UIImageView alloc] initWithImage:playIcon];
+    playIconImageView.backgroundColor = [UIColor blackColor];
+    playIconImageView.frame = CGRectMake(0.0f, 0.0f, size.width, size.height);
+    playIconImageView.contentMode = UIViewContentModeCenter;
+    playIconImageView.clipsToBounds = YES;
+    
+    return playIconImageView;
+}
+
++ (UIImageView *)defaultPlayIconWithThumbnailImage:(UIImage *)thumbnailImage {
+    
+    // same sizing logic with the mediaViewDisplaySize method from JSQMediaItem
+    CGSize size = CGSizeZero;
+    if ([UIDevice currentDevice].userInterfaceIdiom == UIUserInterfaceIdiomPad) {
+        size = CGSizeMake(315.0f, 225.0f);
+    }
+    else {
+        size = CGSizeMake(210.0f, 150.0f);
+    }
+    
+    UIImageView *thumbnailImageView = [[UIImageView alloc] initWithImage:thumbnailImage];
+    thumbnailImageView.backgroundColor = [UIColor blackColor];
+    thumbnailImageView.frame = CGRectMake(0.0f, 0.0f, size.width, size.height);
+    thumbnailImageView.contentMode = UIViewContentModeCenter;
+    thumbnailImageView.clipsToBounds = YES;
+    
+    UIImageView *playIconImageView = [JSQMessagesVideoThumbnailFactory defaultPlayIcon];
+    playIconImageView.backgroundColor = [UIColor clearColor];
+    [thumbnailImageView addSubview:playIconImageView];
+    
+    return thumbnailImageView;
+}
+
+//+ (void)customThumbnailWithMediaItem:(JSQVideoMediaItem *)videoMediaItem
```

Comment:
👍 

I think this is the //only// method we want on this class -- not the two above.

We don't want this to know about the internals of `JSQVideoMediaItem`. 

The completion block here should just return a `UIImage`. (no views)